### PR TITLE
Show unit price in the bulk buy modal

### LINKED ISSUE #7426: Show unit price feature in the bulk buy modal
### Description
Show the unit price information inside the bulk modal: 
Screenshot from 2021-04-15 08-44-58

_NB_: You need to enable group buy for the product from `/admin/products/<product_name>/group_buy_options`.

### Expected Behavior
If feature toggle is enabled for the current user/group/env, it shows the unit price inside this bulk buy modal.

### Actual Behavior
No unit price information is shown inside this bulk buy modal.

### Context
_Originally posted by @sauloperez in https://github.com/openfoodfoundation/openfoodnetwork/issues/6497#issuecomment-820163313_

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?

Show unit price in the bulk buy modal.
Closes #7426

#### What should we test?
You need to enable group buy for the product from `/admin/products/<product_name>/group_buy_options`

##### When `unit_price` feature toggle is enabled
In a shop front, we should display the unit price information near the product (and specially for a _bulk_ one):
Capture d’écran 2021-04-15 à 11 08 36

And when you click on button to add product into cart, the bulk buy modal should open with unit price information:
Capture d’écran 2021-04-15 à 11 08 40

##### When `unit_price` feature toggle is not enabled
Nothing should change, the bulk buy modal should not display the unit price information:
Capture d’écran 2021-04-15 à 11 10 41

#### Release notes
Display unit price information into the bulk but modal.

Changelog Category: User facing changes

### CODE REVIEW

**Comment by Erioldoesdesign** [comment]
Just dropping in with the thumbs up from design 👍

File: app/assets/stylesheets/shared/question-mark-icon.scss [hunk 1]
```diff
@@ -58,6 +56,7 @@
   margin-left: -7.4rem;
   margin-top: -0.1rem;
   background-color: transparent;
+  z-index: 1005;
```
Comment (sauloperez) [inline app/assets/stylesheets/shared/question-mark-icon.scss:59]: is there any science behind this number? I assume so

File: app/assets/javascripts/templates/bulk_buy_modal.html.haml [hunk 2]
```diff
@@ -2,12 +2,21 @@
...
-  .columns.small-6
-    .variant-unit {{ ::variant.unit_to_display }}
-  .columns.small-6
+.flex.variant-bulk-buy-price-summary{style: "justify-content: space-around;"}
+  .variant-unit
+    {{ ::variant.unit_to_display }}
+  .div
     {{ variant.line_item.total_price | localizeCurrency }}
 
+  .unit-price{"ng-if": "showunitprice"}
```
Comment (sauloperez) [inline app/assets/javascripts/templates/bulk_buy_modal.html.haml:11]: it'd be great if we could make this argument more readable like `show_unit_price`. There's quite some code the reader needs to go through so some help is more than welcome.

File: app/assets/stylesheets/shared/question-mark-icon.scss (hunk 1 above)
Comment (jibees) [reply app/assets/stylesheets/shared/question-mark-icon.scss:59]: Yes, modal is `1004`
So, it 1004 + 1

File: app/assets/javascripts/templates/bulk_buy_modal.html.haml (hunk 2 above)
Comment (jibees) [reply app/assets/javascripts/templates/bulk_buy_modal.html.haml:11]: I had some troubles with this. I tried camelCase but it failed. I will try with snake case.

Comment (sauloperez) [reply app/assets/javascripts/templates/bulk_buy_modal.html.haml:11]: :see_no_evil: then I clearly don't know how this works.

File: app/assets/stylesheets/shared/question-mark-icon.scss (hunk 1 above)
Comment (sauloperez) [reply app/assets/stylesheets/shared/question-mark-icon.scss:59]: any chance we could better show our intent with vars?

**Review by andrewpbrett — APPROVED** [review]
LGTM, agree with Pau though that those two things would be nice to address first.

File: app/assets/javascripts/templates/bulk_buy_modal.html.haml (hunk 2 above)
Comment (jibees) [reply app/assets/javascripts/templates/bulk_buy_modal.html.haml]: I've tested with `snake_case`... it fails.
The only solution I found is to mix both... 
Use `showunitprice` and `show_unit_price`...
See commit: a8c9f87dc

File: app/assets/stylesheets/shared/question-mark-icon.scss (hunk 1 above)
Comment (jibees) [reply app/assets/stylesheets/shared/question-mark-icon.scss]: Yes, sure.
See commit d0e123f00

**Review by sauloperez — APPROVED** [review]

**Comment by jibees** [comment]
Before testing, will do some git history cleaning. ;)

**Comment by filipefurtad0** [comment]
Thanks @jibees and @sauloperez for stpotting this missing bit!

Looks good across platforms, see below:

- **Desktop** Ubuntu 20.04, Chrome 89 (left), and Firefox 87 (right):

Peek 2021-04-20 18-10

- **Desktop** Mac, Safari (tested the same functionality as above)

image

- **Desktop** Windows Firefox 
image



- Mobile iPhone - iOS/Safari
image

- Mobile Android - Chrome on Samsung Galaxy S20
image

Checking the test cases you pointed out:

- In a shop front, we should display the unit price information near the product (and specially for a bulk one) - OK
- clicking on button to add product into cart, the bulk buy modal should open with unit price information - OK
- When unit_price feature toggle is not enabled - checked as well, no changes when unit price is not enabled - OK  

Good to go.

PS: there are issues in the pics above related to the known issue #7230.

**Pull request merged**